src/auxiliary/mod.rs

```diff
@@ -0,0 +1,82 @@
+// Copyright 2018 OpenST Ltd.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//!This module is about interaction with auxiliary block chain.
+
+use futures::Future;
+use web3::contract::{Contract, Options};
+use web3::transports::Http;
+use web3::types::{H160, Address, H256};
+use rlp;
+use sha3::{Digest, Keccak256};
+
+use super::ethereum::Ethereum;
+use ethereum::types::{Block, Error, ErrorKind};
+
+const REPORT_BLOCK_ESTIMATED_GAS: i32 = 3_000_000;
+
+/// Reports block on block store if not already reported.
+///
+/// # Arguments
+///
+/// * `blockchain` - A blockchain object.
+/// * `event_loop` - The reactor's event loop to handle the tasks spawned.
+/// * `block_store_address` - The address of origin block store.
+/// * `validator_address` - The address of origin validator address.
```

Comment:
Why origin validator address?
```suggestion
/// * `validator_address` - The address of auxiliary validator address.
/// * `block` - The block to report.
```